front/app/containers/ProjectsShowPage/shared/survey/MicrosoftFormsSurvey.tsx

```diff
@@ -1,23 +1,30 @@
-import React from 'react';
-import styled from 'styled-components';
-import { Box } from 'cl2-component-library';
-
-const StyledIframe = styled.iframe`
-  display: block;
-  height: 600px;
-  width: 100%;
-  border: 1px solid #ccc;
-`;
+import React, { useState } from 'react';
+import { Box, Spinner } from 'cl2-component-library';
 
 type Props = {
   microsoftFormsUrl: string;
   className?: string;
 };
 
 const MicrosoftFormsSurvey = ({ microsoftFormsUrl, className }: Props) => {
+  const [loading, setLoading] = useState(true);
+
   return (
     <Box display="flex" justifyContent="center" className={className}>
-      <StyledIframe src={microsoftFormsUrl} />
+      <Box
+        as="iframe"
+        border="1px solid #ccc"
+        display={loading ? 'none' : 'block'}
+        h="600px"
+        w="100%"
+        src={microsoftFormsUrl}
+        onLoad={() => {
```

Comment:
I've introduced artificial delay for the loading state here because this is how long it takes for the form to "autofocus". I need to make sure it tries to change the focus when the iframe is with `display: none` so that it doesn't have any effect.

Reply:
I dislike this solution as well and completely agree with you! The issue is that the iframe is cross-origin so what you are proposing (and some other approaches I tried as well) result in:
```
 DOMException: Blocked a frame with origin "http://localhost:3000" from accessing a cross-origin frame.
```
If it makes you feel any better, I tested this timeout on multiple browsers and with different connection speeds so although I can't guarantee it will work 100% of the time, I'm confident for the vast majority of cases it should be fine.